modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -571,8 +576,25 @@ char** iocsh_attempt_completion(const char* word, int start, int end)
             // known argument type
             rl_attempted_completion_over = 1;
 
+            //if input string has a ., we know what record it is.
+            std::string str = word;
+            std::string init_field = "";
+            std::string pv = "";
+
+            size_t dot = str.find(".");
+
+            if (dot != std::string::npos) {
+                pv = str.substr(0, dot);
+                init_field = str.substr(dot+1u, str.length());
+            }
+
+            if(!init_field.empty()) {
+                return iocshCompleteField(pv.c_str(), init_field.c_str());
+            }
+
+            //Q: where is iocshCompleteRecord instantiated?
             if(argdef->type==iocshArgStringRecord && iocshCompleteRecord) {
-                return (*iocshCompleteRecord)(word);
+                return iocshCompleteRecord(word);
```

Comment:
Does this actually work? Because `iocshCompleteRecord` is a pointer to a function, so you would need to dereference it first to call the actual function.  
See: https://github.com/epics-base/epics-base/blob/5eff3803a83fbc3ea855b63841f5359c09830ade/modules/database/src/ioc/db/dbIocRegister.c#L506

Reply:
Yes, I can't quite remember who advised me on this, but effectively in c++ you can call function pointers in the same way you call regular functions. I've tested commands in the shell and it seems to work just fine.

Reply:
It works in C code too, there are implicit conversions defined by the language and we use this elsewhere in the IOC code. The Pointers to functions section of [this page](https://en.cppreference.com/w/c/language/pointer) is a good place to start if you want to read all about it.

Reply:
The current code is my personal style to make it clear that there is a pointer de-reference here.  Otherwise someone reading later (eg. me) can miss that this is not a regular function call.  Not a big deal in this case as the NULL check on the previous line should be a clue (surely I wouldn't miss that!).